006-graphql-pagination.md

```diff
@@ -0,0 +1,84 @@
+# GraphQL Pagination
+
+We should use [Relay Cursor Connections](https://facebook.github.io/relay/graphql/connections.htm#) to paginate results in GraphQL. 
+
+**Discussion:** https://github.com/DoSomething/rfcs/pull/6
+
+## Problem
+
+We currently return plain lists, like `[Post]`, for paginated GraphQL queries:
+
+```gql
+{
+  posts(count: 10, page: $page) {
+    id
+    type
+    url
+  }
+}
+```
+
+This gives us no way to know whether there's another page of results without making a request to check. Because we built these on top of standard `OFFSET` pagination, there are also some performance issues with larger datasets & the potential for duplicate results to appear (described in more detail under "Alternatives").
+
+## Proposal
+
+We should use [Relay Cursor Connections](https://facebook.github.io/relay/graphql/connections.htm#) to paginate results in GraphQL, and add REST API support to paginate via cursors (e.g. "get me all the results after #1015" rather than "get me the tenth page of results") to Northstar and Rogue.
+
+```gql
+{
+  paginatedPosts(first: 10, after: $cursor) {
+    edges {
```

Comment:
should this be `posts`?

Reply:
This is the "standard" `{ edges { node, cursor }, pageInfo }` structure from the [GraphQL Cursor Connection](https://facebook.github.io/relay/graphql/connections.htm) spec. The terminology is a little awkward, but it's nice that it allows us to handle pagination logic the same way no matter what we're paginating!

Reply:
Thanks! I was looking there and they only have the one example similar to this at the top - in this example, is a `node` a `post`?

Reply:
Yep! In this example, each `node` would be a `Post`.

Reply:
Got it! Thank you!!!